File: Proyecto1 AS/app/src/main/java/com/example/app1_iniciocierre/model/Usuario.kt
package com.example.app1_iniciocierre.model

data class Usuario(
    val id: Int,
    val username: String,
    val rol: Rol
)

enum class Rol(val etiqueta: String) {
    ADMINISTRADOR("Administrador"),
    AUDITOR("Auditor"),
    CLIENTE("Cliente")
}

fun rolParaUsuario(id: Int): Rol = when (id) {
    1, 2 -> Rol.ADMINISTRADOR
    3 -> Rol.AUDITOR
    else -> Rol.CLIENTE
}

File: Proyecto1 RC/src/model/Usuario.ts
import { Rol, ROLES } from './Rol';

export interface DatosUsuario {
  id: number;
  username: string;
  nombre: string;
  email: string;
  rol: Rol;
}

export class Usuario {
  readonly rol: Rol;

  constructor(
    private readonly _id: number,
    private readonly _username: string,
    private readonly _nombre: string,
    private readonly _email: string,
  ) {
    this.rol = Usuario.getRolPorId(_id);
  }

  get id(): number {
    return this._id;
  }

  get username(): string {
    return this._username;
  }

  get nombre(): string {
    return this._nombre;
  }

  get email(): string {
    return this._email;
  }

  static getRolPorId(id: number): Rol {
    if (id === 1 || id === 2) {
      return ROLES.ADMINISTRADOR;
    }
    if (id === 3) {
      return ROLES.AUDITOR;
    }
    return ROLES.CLIENTE;
  }

  toObject(): DatosUsuario {
    return {
      id: this._id,
      username: this._username,
      nombre: this._nombre,
      email: this._email,
      rol: this.rol,
    };
  }

  static desdeObject(data: DatosUsuario): Usuario {
    return new Usuario(data.id, data.username, data.nombre, data.email);
  }
}
